src/dso_api/dynamic_api/management/commands/dump_serializers.py

```diff
@@ -0,0 +1,266 @@
+            f"# {app.verbose_name or app.app_label}.{model._meta.verbose_name}\n\n\n"
+        )
+
+    def write_sub_serializers(self, serializer: DSOSerializer, seen_serializers: set):
+        """Write the dependant serializers"""
+        for name, field in serializer.fields.items():
+            if isinstance(field, serializers.Serializer):
+                # Make sure generic serializers are not reprinted each time
+                if field.__class__ in seen_serializers:
+                    continue
+
+                seen_serializers.add(field.__class__)
+
+                # First write any sub-dependant fields (e.g. _links section items)
+                self.write_sub_serializers(field, seen_serializers)
+
+                # Then write the serializer as any other regular serializer
+                self.write_serializer(field)
+
+    def write_serializer(self, serializer: DSOSerializer):
+        """Write the representation of the serializer"""
+        serializer_class = serializer.__class__
+        bases = ", ".join(base_class.__name__ for base_class in serializer_class.__bases__)
+        self.stdout.write(f"class {serializer_class.__name__}({bases}):\n")
+
+        if serializer_class.__doc__:
+            self.write_docstring(serializer_class.__doc__)
+
+        self.write_factory_docs(serializer_class)
+
+        # Split fields in auto created vs declared fields
+        fields = dict(serializer.fields)
+        declared_names = sorted(serializer._declared_fields.keys())
+        for name in declared_names:
+            field = fields.pop(name, None)
+            if field is not None:  # e.g. "schema" field overwritten by subclass.
+                self.write_field(field)
+
+        if fields:
+            self.stdout.write("\n    # Auto created fields during each request:\n")
+            for field in fields.values():
+                self.write_field(field)
+
+        if hasattr(serializer_class, "Meta"):
+            self.stdout.write("\n")
+            self.write_serializer_meta(serializer_class)
+
+        self.stdout.write("\n\n")
+
+    def write_docstring(self, doc: str) -> None:
+        """Write the docstring for a class."""
+        if "\n" not in doc:  # if not formatted already
+            # Wrap our description text
+            doc = textwrap.fill(doc, 96, subsequent_indent="    ")
+            if "\n" in doc:  # if wrapped
+                doc += "\n"
+        self.stdout.write(f'    """{doc}"""\n\n')
+
+    def write_factory_docs(self, serializer_class: Type[DynamicSerializer]) -> None:
+        """Write debugging information about the factory function."""
+        # Write some factory information
+        factory_name = self._get_serializer_factory_name(serializer_class)
+        if issubclass(serializer_class, DynamicSerializer):
+            self.stdout.write(f"    # Set by {factory_name}():\n")
+            self.stdout.write(f"    # table_schema = {serializer_class.table_schema}\n")
```

Comment:
Minor nitpicking. You seem to be doing identation manually. That soon starts to become errorprone, do I have 7 or 8 spaces (=level 2), 11 or 12 (=level 3), etc. You could do something like this:

```python
ident = lambda level: " " * 4 * level
...
self.stdout.write(f"{ident(1)}# Set by {factory_name}():\n")
self.stdout.write(f"{ident(1)}# table_schema = {serializer_class.table_schema}\n")
...
... 
self.stdout.write(f"{ident(2)}{name} = {value!r}\n")
```
Looks weird, but it is immediately clear what indentation level you are on.

Reply:
Less verbose:

```python
def indented(level, s):
    self.stdout.write(" " * 4 * level)
    self.stdout.write(s)

Reply:
@lbam That's a much better solution. Thanks!

Reply:
well in this case (and with the dump_models) command, there is only one indentation level. This won't change either so I'd vote for keeping the code as it is